controllers/custodian/config.go

```diff
@@ -39,7 +40,11 @@ type Config struct {
 	// Workers denotes the number of worker threads for the custodian controller.
 	Workers int
 	// SyncPeriod is the duration after which re-enqueuing happens.
+	// Deprecated: this field will be removed in the future. Please use RequeueInterval instead.
+	// TODO (shreyas-s-rao): remove this in druid v0.22
 	SyncPeriod time.Duration
+	// RequeueInterval is the interval for custodian controller to requeue reconciliations.
+	RequeueInterval time.Duration
```

Comment:
I would recommend that we keep `SyncPeriod` for now and not introduce anything else for the moment. We need to look at what we expose carefully and we can decide when we do v1beta1 of the API. Both `SyncPeriod` and `RequeueInterval` fields are very generic and do not tell much on what they will be used for. One needs to read the comment to understand what it means.

In this PR let us just resolve one bug and be done with it.